pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -464,6 +466,17 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	// needed for luksOpen synchronization when devices are encrypted and the osd is prepared with LVM
 	hostIPC := osdProps.storeConfig.EncryptedDevice || osdProps.encrypted
 
+	osdLabels := c.getOSDLabels(osd, failureDomainValue, osdProps.portable)
+	clusterIP, err := c.createOSDService(osd, osdLabels)
+	if err != nil {
+		logger.Errorf("failed to configure osd service for osd.%d. %v", osd.ID, err)
```

Comment:
We'll want to fail the reconcile in this case instead of just logging, which was likely from my early prototype.

Reply:
fixed.